### PR TITLE
Send appname as header on DCL upload

### PR DESCRIPTION
**Pull request by d047491**

### CODE REVIEW

File: pkg/supply/supply.go [hunk 1]
```diff
@@ -280,12 +280,19 @@ func (s *Supplier) upload(creds *services.IASCredentials, tlsCfg tlsConfig, root
...
+	}
 	u := uploader.Uploader{
 		Log:           s.Log,
 		Root:          path.Join(s.Stager.BuildDir(), rootDir),
 		Client:        client,
 		AMSInstanceID: creds.AmsInstanceID,
-		UserAgent:     fmt.Sprintf("cloud-authorization-buildpack/%s", s.BuildpackVersion),
+		ExtraHeaders: map[string]string{
+			"User-Agent":   fmt.Sprintf("cloud-authorization-buildpack/%s", s.BuildpackVersion),
+			"X-Cf-Appname": vcapApp.ApplicationName,
```
Comment (f-blass) [inline pkg/supply/supply.go:294]: Let's use a more generic name that also fits for other use cases, e.g. if we provide some functionality to upload via CAP or as a k8s job. E.g. just remove the _cf_ part? `X-App-Name` or `X-Appname` ?

File: pkg/supply/env/config.go [hunk 2]
```diff
@@ -20,6 +20,35 @@ type amsDataDeprecated struct {
...
+	Limits          struct {
+		Fds int `json:"fds"`
+	} `json:"limits"`
+	Name             string   `json:"name"`
+	OrganizationID   string   `json:"organization_id"`
+	OrganizationName string   `json:"organization_name"`
+	SpaceID          string   `json:"space_id"`
+	SpaceName        string   `json:"space_name"`
+	Uris             []string `json:"uris"`
+	Users            any      `json:"users"`
```
Comment (f-blass) [inline pkg/supply/env/config.go:37]: From what I can see, only the `ApplicationName` is used in the coding. Let's strip down the struct to what we need exactly, no need to parse the other values.

File: pkg/supply/env/config.go [hunk 3]
```diff
@@ -20,6 +20,35 @@ type amsDataDeprecated struct {
...
+}
+
+func LoadVcapApplication(log *libbuildpack.Logger) (VcapApplication, error) {
```
Comment (f-blass) [inline pkg/supply/env/config.go:40]: This func always returns error = nil. Remove error return value?

**Review by f-blass — CHANGES_REQUESTED** [review]

**Comment by d047491** [comment]
commited suggested changes

**Review by f-blass — APPROVED** [review]

**Pull request merged**